plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -149,6 +157,49 @@ public void writeBlob(String blobName, InputStream inputStream, long blobSize, b
         });
     }
 
+    @Override
+    public boolean isMultiStreamUploadSupported() {
+        return blobStore.isMultipartUploadEnabled();
+    }
+
+    @Override
+    public boolean isRemoteDataIntegritySupported() {
+        return true;
+    }
+
+    @Override
+    public CompletableFuture<Void> writeBlobByStreams(WriteContext writeContext) throws IOException {
+        UploadRequest uploadRequest = new UploadRequest(
+            blobStore.bucket(),
+            buildKey(writeContext.getFileName()),
+            writeContext.getFileSize(),
+            writeContext.getWritePriority(),
+            writeContext.getUploadFinalizer(),
+            writeContext.doRemoteDataIntegrityCheck(),
+            writeContext.getExpectedChecksum()
+        );
+        try {
+            long partSize = blobStore.getAsyncUploadUtils().calculateOptimalPartSize(writeContext.getFileSize());
+            StreamContext streamContext = SocketAccess.doPrivileged(() -> writeContext.getStreamProvider(partSize));
+            try (AmazonAsyncS3Reference amazonS3Reference = SocketAccess.doPrivileged(blobStore::asyncClientReference)) {
+
+                S3AsyncClient s3AsyncClient = writeContext.getWritePriority() == WritePriority.HIGH
+                    ? amazonS3Reference.get().priorityClient()
+                    : amazonS3Reference.get().client();
+                CompletableFuture<Void> returnFuture = new CompletableFuture<>();
```

Comment:
Yes, you are right. We don't need this anymore. This was there earlier because we were doing few more checks on completableFuture's completion. @raghuvanshraj